content/blog/cicd-pipelines-with-codefresh-and-pulumi/index.md

```diff
@@ -0,0 +1,66 @@
+
+## Building Modern Infrastructure
+
+Just like your application code, you can build your infrastructure with a CI/CD pipeline. Pulumi lets developers write infrastructure using their favorite language, such as Typescript, Python, or go.  Developers can take advantage of abstractions, code reuse, refactoring, favorite development environments, and testing tools.
+
+In addition to supporting multiple languages, you can build infrastructure on any cloud such as AWS, Azure, GCP, and Kubernetes and take full advantage of their service offerings. Choose your toolchain and frameworks, and you're set to deploy infrastructure on any major cloud provider and even hybrid and on-premises environments.
+
+By leveraging infrastructure as code combined with software developement practices, engineers can achieve greater productivity while enabling their development teams to be more "self-serve" with appropriate policies and guardrails in place. Pulumi brings cloud security engineering to the infrastructure and development teams. Using one common workflow that spans any cloud environment, whether public, private, or hybrid, security teams can enforce policies that accomplish cloud governance projects of all kinds — security, compliance, cost control, and more.
+
+To see how infrastructure as code simplifies infrastructure deployment, let's look at an example for deploying Kubernetes with python on AWS. The full [example](https://github.com/pulumi/examples/tree/master/aws-py-eks) is available on Github.
+
+The code breaks up the Kubernetes deployment into python modules:
+
+- [vpc.py](https://github.com/pulumi/examples/blob/master/aws-py-eks/vpc.py) creates an AWS Virtual Private Cloud containing the cluster, security group, and networking.
+- [iam.py](https://github.com/pulumi/examples/blob/master/aws-py-eks/iam.py) creates roles for the cluster, and the nodegroup and policy attachments
+- [utils.py](https://github.com/pulumi/examples/blob/master/aws-py-eks/utils.py) creates a `kubeconfig` file that can be used with `kubectl` to manage Kubernetes
+
+The main program (__main__.py) creates the Kubernetes cluster by calling the modules to deploy a cluster and a nodegroup with the policies applied. Using modules, we can make changes to the vpc, networking, roles, and policies and customize it to our application. In a CI/CD, all these changes would be pushed through a pipeline before deployment.
+
+Although this example is written in Python for AWS, there are similar examples in other languages and cloud providers in [Pulumi Examples](https://github.com/pulumi/examples).
+
+## Putting it Together
+
+This article described how we could build both applications and infrastructure with CI/CD pipelines. To learn more about how CodeFresh and Pulumi can automate Kubernetes deployments join us for a [technical session](https://www.pulumi.com/resources/ci-cd-pipelines-for-kubernetes-apps-with-codefresh/) on January 21st.
```

Comment:
George had a suggestion (from the email thread) about moving this to the top. Maybe it could just simply say `Join us for a [technical session](link) ...`.

Reply:
Added CTA to before the cut line